.ci/post_review.py

```diff
@@ -0,0 +1,214 @@
+#! usr/bin/env python3
+
+# Automatically create review from clang-tidy report
+# Copyright (c) Ashar Khan <ashar786khan.at.gmail.com>
+
+
+# Boost Software License 1.0
+# Permission is hereby granted, free of charge, to any person or
+# organization obtaining a copy of the software and accompanying
+# documentation covered by this license (the "Software") to use,
+# reproduce, display, distribute, execute, and transmit the Software,
+# and to prepare derivative works of the Software, and to permit
+# third-parties to whom the Software is furnished to do so, all
+# subject to the following:
+
+# The copyright notices in the Software and this entire statement,
+# including the above license grant, this restriction and the following
+# disclaimer, must be included in all copies of the Software, in whole
+# or in part, and all derivative works of the Software, unless such copies
+# or derivative works are solely in the form of machine-executable object
+# code generated by a source language processor.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE, TITLE AND NON-INFRINGEMENT. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDERS OR ANYONE DISTRIBUTING THE SOFTWARE BE LIABLE FOR ANY
+# DAMAGES OR OTHER LIABILITY, WHETHER IN CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+
+import argparse
+from github import Github
+import unidiff
+import os
+import requests
+
+
+def fetch_diff(repository, pr, token):
+    headers = {
+        "Accept": "application/vnd.github.v3.diff",
+        "Authorization": f"token {token}",
+    }
+    url = f"https://api.github.com/repos/{repository}/pulls/{pr}"
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
+
+    return unidiff.PatchSet(response.text)
+
+
+def parse_report_file(content):
+    root_path = os.getcwd()
+    parsed_report = []
+    n = len(content)
+    i = 0
+    while (i < n):
+        if "error" in content[i] or "warning" in content[i]:
+            abs_file_path, line_number, _, severity, message = content[i].split(
+                ":", maxsplit=4)
+            relative_path = os.path.relpath(abs_file_path, root_path)
+
+            k = 1
+            body = ""
+            while i+k < n and not ("error" in content[i+k] or "warning" in content[i+k]):
+                body += "\n" + content[i +
+                                       k].replace(abs_file_path, relative_path)
+                k += 1
+            i += k
+
```

Comment:
I am not a python expert, but are the line breaks wanted?

Reply:
```suggestion
            abs_file_path, line_number, _, severity, message = content[i].split(
                ":", maxsplit=4)
            relative_path = os.path.relpath(abs_file_path, root_path)

            k = 1
            body = ""
            while i+k < n and not ("error" in content[i+k] or "warning" in content[i+k]):
                body += "\n" + content[i + k].replace(abs_file_path, relative_path)
                k += 1
            i += k

```

I ran VSCode Format document for this file and it made this change.